datadog/tests/resource_datadog_integration_slack_channel_test.go

```diff
@@ -0,0 +1,177 @@
+package test
+
+import (
+	"fmt"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	communityClient "github.com/zorkian/go-datadog-api"
+)
+
+func TestAccDatadogIntegrationSlackChannel_Basic(t *testing.T) {
+	accProviders, clock, cleanup := testAccProviders(t, initRecorder(t))
+	reg, _ := regexp.Compile("[^a-zA-Z0-9]+")
+	uniqueChannelTeamName := reg.ReplaceAllString(uniqueEntityName(clock, t), "")
+	defer cleanup(t)
+	accProvider := testAccProvider(t, accProviders)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    accProviders,
+		CheckDestroy: testAccCheckDatadogIntegrationSlackChannelDestroy(accProvider),
+		Steps: []resource.TestStep{
+			{
+				// Workaround to ensure we create the slack integration before running the tests.
+				Config: emptyLogsArchiveConfig(),
```

Comment:
nit, can we use a datasource here instead. Feels generally safer, though logs_archive_order may be fine.

Reply:
Could we move this method call to the PreCheck instead actually?

Reply:
I believe the provider is initialized in the steps which is performed after the PreCheck so we might not have access to the datadog provider. Will double check

Reply:
Good idea on the data source but unfortunately we can't use `PreCheck` in this case. It seems PreCheck happens before the provider is resolved.

Reply:
You should be able to pass the `accProvider` to the `PreCheck` function via closure.

Reply:
Gave the above a shot and ran into: 
```
panic: interface conversion: interface {} is nil, not *datadog.ProviderConfiguration [recovered]
```
Although we initialize `accProvider`, it seems that the actual provider are loaded in during the test steps only. The accProvider seems to be just a map that is used to check what providers will be used in the tests, e.g. `datadog`.

Looking at the code PreCheck itself seems to be run before the providers are resolved: https://github.com/hashicorp/terraform-plugin-sdk/blob/master/helper/resource/testing.go#L542-L544

Based on what I can tell, within Config step is the first place where the provider gets loaded in

Reply:
Yeah precheck is a no go, I tried it too a while ago and couldn't haha.
Do you have to use an actual resource though ? What happens if you provide an empty string ?

Reply:
Tested it out and empty string will not work. Looks like terraform sdk expects strictly a resource in the config to load in the provider